app/Models/User.php

```diff
@@ -206,6 +207,20 @@ public function atc(): Attribute
         );
     }
 
+    /**
+     * Returns a boolean value whether the pilot is actively flying or not.
+     *
+     * @return Attribute
+     */
+    public function isFlying(): Attribute
+    {
+        return Attribute::make(
+            get: function ($_, $attrs) {
+                return (bool) Pirep::where(['user_id' => $this->id, 'state' => PirepState::IN_PROGRESS])->count();
```

Comment:
I would change this to use `latest('modified_at')`. In case there is a bugged/old PIREP.

Reply:
Maybe the name also to `inFlight()`. Since you're checking if they're in a particular state. Then the same property can probably be added to `Aircraft` later.

Reply:
You can also, instead of `count()`, use `exists()`. Then you don't need to cast it. And you can add more constraints, to say in the last day, etc. But I think latest on when it's been modified should be ok. There will be ongoing updates to that record.